interval/interval.go

```diff
@@ -15,9 +15,38 @@
 // Package interval provides a struct representing a time interval.
 package interval
 
-import "time"
+import (
+	"math/rand"
+	"time"
+)
 
 // Interval represents the interval [Start, End).
 type Interval struct {
 	Start, End time.Time
 }
+
+// RandomInstant returns a random time (to second precision) that falls within
+// the Interval.
+//
+// If the Interval is nil, or Interval.End <= Interval.Start, the zero value
+// time.Time will be returned.
+func (i *Interval) RandomInstant() time.Time {
+	if i == nil {
+		return time.Time{}
+	}
+
+	if i.Start.Equal(i.End) || i.Start.After(i.End) {
```

Comment:
if `i.Start == i.End` then wouldn't returning i.EitherOfThem be good enough? It is as random as you can get within a zero-width interval

Reply:
No - if you look at the documentation for the Interval type, the interval is [Start, End), so if Start == End there is nothing in the interval/the interval is invalid (how do you include Start but exclude End when they are the same?).

Reply:
I overlooked that it is [start, end) - ok then.